src/index.js

```diff
@@ -0,0 +1,194 @@
+import { createElement, Component } from 'react';
+import generateOutsideCheck from './generateOutsideCheck';
+
+/**
+ * A higher-order-component for handling onClickOutside for React components.
+ */
+const registeredComponents = [];
+const handlers = [];
+
+/**
+ * This function generates the HOC function that you'll use
+ * in order to impart onOutsideClick listening to an
+ * arbitrary component. It gets called at the end of the
+ * bootstrapping code to yield an instance of the
+ * onClickOutsideHOC function defined inside setupHOC().
+ */
+export default function onClickOutsideHOC(WrappedComponent, config) {
+  return class onClickOutside extends Component {
+    static displayName = `OnClickOutside(${ WrappedComponent.displayName || WrappedComponent.name || 'Component' })`
+
+    static defaultProps = {
+      eventTypes: ['mousedown', 'touchstart'],
+      excludeScrollbar: (config && config.excludeScrollbar) || false,
+      outsideClickIgnoreClass: 'ignore-react-onclickoutside',
+      preventDefault: false,
+      stopPropagation: false,
+    }
+
+    static getClass = () => WrappedComponent.getClass ? WrappedComponent.getClass() : WrappedComponent
+
+    /**
+     * Access the WrappedComponent's instance.
+     */
+    getInstance() {
+      return WrappedComponent.prototype.isReactComponent ? this.instanceRef : this;
```

Comment:
isnt on the `master` and ive tried to focus the PR on refactoring to es6 class only, gonna add it though